crates/lune-roblox/src/datatypes/types/unique_id.rs

```diff
@@ -0,0 +1,93 @@
+use std::fmt;
+
+use lune_utils::TableBuilder;
+use mlua::BString;
+use mlua::prelude::*;
+
+use super::super::*;
+use crate::exports::LuaExportsTable;
+
+use rbx_dom_weak::types::UniqueId as DomUniqueId;
+
+/// An implementation of the `UniqueId` Robox datatype.
+/// This type is not exposed to users in engine, but is used as an ID for
+/// Instances by the engine.
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub struct UniqueId {
+    id: u128,
+}
+
+impl LuaExportsTable for UniqueId {
+    const EXPORT_NAME: &'static str = "UniqueId";
+
+    fn create_exports_table(lua: mlua::Lua) -> mlua::Result<mlua::Table> {
+        let from_string = |_: &Lua, input: BString| {
+            if input.len() == 16 {
+                let mut bytes = [0; 16];
+                for n in 0..16 {
+                    bytes[n] = input[n];
+                }
```

Comment:
We can probably use `copy_from_slice` on the array to make this a bit neater?

Reply:
Right you are! I forgot that method exists.